lib/game.rb

```diff
@@ -0,0 +1,201 @@
+class Game
+  def initialize(board, players)
+    @board = board
+    @players = players
+  end
+
+  private
+
+  def clear_board(_board)
+    board = [' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ']
+  end
+
+  def player_turn(board, index, player)
+    param = ''
+
+    if player === @players[0]
+      param = 'X'
+    elsif player === @players[1]
+      param = 'O'
+    end
+
+    case index
+    when 0
+      board[0] = param
+    when 1
+      board[1] = param
+    when 2
+      board[2] = param
+    when 3
+      board[3] = param
+    when 4
+      board[4] = param
+    when 5
+      board[5] = param
+    when 6
+      board[6] = param
+    when 7
+      board[7] = param
+    when 8
+      board[8] = param
+    end
+  end
+
+  def position_filled?(board, position)
+    if board[position] == ' ' || board[position] == '' || board[position].nil?
+      false
+    else
+      true
+    end
+  end
+
+  def in_range(move)
+    loop do
+      unless move.between?('0', '8')
+        puts 'Please enter a number in range of 0-8'
+        move = gets.chomp
+      end
+      break if move.between?('0', '8')
+    end
+    move.to_i
+  end
+
+  def valid?(move)
+    loop do
+      if move.class != Integer || !move.between?(0, 8)
+        puts 'Invalid move, enter number in range of 0-8'
+        move = gets.chomp
+        move = in_range(move)
+      end
+      break if move.class == Integer && move.between?(0, 8)
+    end
+    move
+  end
+
+  def check_position(board, move, player)
+    loop do
+      if position_filled?(board, move)
+        puts puts "#{player.name}, that position is filled, Try again..."
+        move = gets.chomp
+        move = in_range(move)
+      end
+      break unless position_filled?(board, move)
+    end
+    move
+  end
+
+  def won(board)
+    p1_move = []
+    p2_move = []
+
+    first_cmbn = [0, 1, 2]
+    second_cmbn = [3, 4, 5]
+    third_cmbn = [6, 7, 8]
+    fourth_cmbn = [0, 3, 6]
+    fifth_cmbn = [1, 4, 7]
+    sixth_cmbn = [2, 5, 8]
+    seventh_cmbn = [0, 4, 8]
+    eighth_cmbn = [2, 4, 6]
+
+    board.each_with_index do |i, index|
+      p1_move << index if i == 'X'
+      p2_move << index if i == 'O'
+    end
+
+    if (first_cmbn - p1_move).empty? || (second_cmbn - p1_move).empty? || (third_cmbn - p1_move).empty? || (fourth_cmbn - p1_move).empty? || (fifth_cmbn - p1_move).empty? || (sixth_cmbn - p1_move).empty? || (seventh_cmbn - p1_move).empty? || (eighth_cmbn - p1_move).empty?
+      1
+    elsif (first_cmbn - p2_move).empty? || (second_cmbn - p2_move).empty? || (third_cmbn - p2_move).empty? || (fourth_cmbn - p2_move).empty? || (fifth_cmbn - p2_move).empty? || (sixth_cmbn - p2_move).empty? || (seventh_cmbn - p2_move).empty? || (eighth_cmbn - p2_move).empty?
+      2
+    else
```

Comment:
There is an easy way to refactor it. Think about 2d array. The solution is in there.